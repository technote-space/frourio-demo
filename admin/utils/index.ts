import { singular, plural } from 'pluralize';

export const getWord = (word: string, number: number): string => number <= 1 ? singular(word) : plural(word);
