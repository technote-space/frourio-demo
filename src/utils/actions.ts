import type { UseToastOptions } from '@chakra-ui/react';
import type { PageKeys } from '~/_pages';

type Dispatch = ((value: { type: string; [key: string]: any }) => void);

export const setAdmin = (dispatch: Dispatch, admin: { name?: string; icon?: string }) => dispatch({
  type: 'SET_ADMIN',
  admin,
});

export const openSidebar  = (dispatch: Dispatch) => dispatch({ type: 'OPEN_SIDEBAR' });
export const closeSidebar = (dispatch: Dispatch) => dispatch({ type: 'CLOSE_SIDEBAR' });

export const changePage = (dispatch: Dispatch, page: PageKeys) => dispatch({ type: 'PAGE', page });
export const logout     = (dispatch: Dispatch) => dispatch({ type: 'LOGOUT' });

export const createToast        = (dispatch: Dispatch, option: UseToastOptions) => dispatch({
  type: 'TOAST',
  toasts: option,
});
export const createWarningToast = (dispatch: Dispatch, option: UseToastOptions) => createToast(dispatch, {
  ...option,
  status: 'warning',
});
export const createErrorToast   = (dispatch: Dispatch, option: UseToastOptions) => createToast(dispatch, {
  ...option,
  status: 'error',
});
