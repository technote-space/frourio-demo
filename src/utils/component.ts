import { FC } from 'react';
import { AuthenticatedPageProps } from '~/components/AuthenticatedPage';

export const getDisplayName = (prefix: string, WrappedComponent: FC<AuthenticatedPageProps>) => `${prefix}(${WrappedComponent.displayName || WrappedComponent.name || 'Component'})`;
