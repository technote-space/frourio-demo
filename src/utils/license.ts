import type { License } from '~/types';
import licenses from '../../license.json';

export const getLicenseList = (): Array<License> => {
  return Object.values(licenses).map(license => ({
    name: license.name,
    version: license.version,
    licenses: typeof license.licenses === 'string' ? license.licenses : license.licenses.join(', '),
    repository: license.repository,
    licenseText: license.licenseText,
  }));
};
