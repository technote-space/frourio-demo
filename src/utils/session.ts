export const getSession = async <T>(key: string, generator: () => Promise<T | undefined>, setter?: (value: any) => void): Promise<T | undefined> => {
  const value = sessionStorage.getItem(key);
  if (null === value) {
    const result = await generator();
    if (undefined !== result && setter) {
      setter(result);
    }

    return result;
  }

  return JSON.parse(value);
};
export const setSession = (key: string, value: any): void => {
  sessionStorage.setItem(key, JSON.stringify(value));
};

// export const getAuthTokenCache   = async(generator: () => Promise<string | undefined>) => getSession(`authToken`, generator, setAuthTokenCache);
// export const setAuthTokenCache   = (value: string) => setSession(`authToken`, value);
// export const clearAuthTokenCache = () => sessionStorage.removeItem(`authToken`);
