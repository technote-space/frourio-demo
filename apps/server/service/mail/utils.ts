import type { MailOptions, MailSettings, MailAddress } from '$/types';
import type { Primitive } from '@frourio-demo/types';
import { htmlToText } from 'html-to-text';
import { fork } from 'child_process';
import { FRONT_URL, SMTP_BCC, SMTP_FROM, SMTP_HOST, SMTP_PASS, SMTP_PORT, SMTP_SECURE, SMTP_USER } from '$/service/env';
import { replaceVariables } from '@frourio-demo/utils/string';
import HeadTemplate from './templates/Head.html';
import HeaderTemplate from './templates/Header.html';
import FooterTemplate from './templates/Footer.html';

export const getSmtpOptions = (): MailOptions => ({
  host: SMTP_HOST,
  port: SMTP_PORT,
  secure: SMTP_SECURE,
  auth: {
    user: SMTP_USER,
    pass: SMTP_PASS,
  },
  tls: { rejectUnauthorized: false },
});
export const getText = (text: string | undefined, html: string): string => text ?? htmlToText(html);
export const getMailSettings = ({ to, bcc, subject, html, text }: {
  to: MailAddress;
  bcc?: MailAddress;
  subject: string;
  html: string;
  text?: string;
}): MailSettings => ({
  from: SMTP_FROM,
  to,
  bcc,
  subject,
  html,
  text: getText(text, html),
});

export const getTemplateVariables = () => ({
  'head': HeadTemplate,
  'header': HeaderTemplate,
  'footer': FooterTemplate,
});
export const getCommonVariables = () => ({
  'logo_url': `${FRONT_URL}/favicon.png`,
  'top_link': `${FRONT_URL}`,
  'terms_link': `${FRONT_URL}/terms`,
  'privacy_link': `${FRONT_URL}/privacy`,
  'contact_link': `${FRONT_URL}/contact`,
});

export const send = async(options: MailOptions, settings: MailSettings): Promise<boolean> => {
  const child = fork(
    'tasks/mail.js',
    [],
    { execArgv: ['-r', 'ts-node/register'] },
  );
  return child.send({ options, settings });
};

export const sendHtmlMail = async(to: MailAddress, subject: string, template: string, variables?: Record<string, Primitive>): Promise<boolean> => send(getSmtpOptions(), getMailSettings({
  to,
  bcc: SMTP_BCC,
  subject: `【Frourioの宿】${subject}`,
  html: replaceVariables(replaceVariables(template, getTemplateVariables()), {
    ...getCommonVariables(),
    ...variables,
  }),
}));
