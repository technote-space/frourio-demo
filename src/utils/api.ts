/* eslint-disable @typescript-eslint/no-explicit-any */
import type { AspidaResponse } from 'aspida';
import type { ApiInstance } from '$/api/$api';
import type { MaybeUndefined } from '~/types';
import aspida from '@aspida/axios';
import { AxiosError } from 'axios';
import { singular } from 'pluralize';
import api from '$/api/$api';
import { changePage, setError } from '~/utils/actions';

export const client = api(aspida());

export const isAxiosError = (target: any): target is AxiosError => {
  return typeof target === 'object' &&
    'config' in target &&
    'request' in target &&
    'response' in target;
};

export const handleAuthError = async <T, U, V, API extends (...args: Array<any>) => Promise<AspidaResponse<T, U, V>>>(
  dispatch: ((value: { type: string; [key: string]: any }) => void),
  fallback: T,
  api: API,
  ...option: Parameters<API>
): Promise<T | MaybeUndefined<T>> | never => {
  try {
    const result = await api(...option);
    console.log(result);

    return result.body;
  } catch (error) {
    console.log(error);
    if (isAxiosError(error)) {
      console.log(error.response);
      if (error.response && 401 === error.response.status) {
        setError(dispatch, error.response.data.message);
        changePage(dispatch, 'logout');
        if (!fallback) {
          return undefined as MaybeUndefined<T>;
        }

        return fallback;
      }

      if (error.response) {
        setError(dispatch, error.response.data.message);
        throw error;
      }
    }

    setError(dispatch, error.message);
    throw error;
  }
};

export type DataTableApi = {
  get: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  post: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  detail: (id: number) => {
    patch: (option?: any) => Promise<AspidaResponse<any, any, any>>;
    delete: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  }
};
export type DataTableApiModels = {
  [key in keyof ApiInstance]: ApiInstance[key] extends {
    get,
    post,
  } ? key : never
}[keyof ApiInstance]
export const getDataTableApi = (model: DataTableApiModels): DataTableApi => ({
  ...client[model],
  detail: id => ({
    patch: client[model][`_${singular(model)}Id`](id).patch,
    delete: client[model][`_${singular(model)}Id`](id).delete,
  }),
});

export const processUpdateData = <T extends Partial<{
  updatedAt,
  createdAt,
}>>(data: T): Omit<T, 'updatedAt' | 'createdAt'> => {
  delete data.updatedAt;
  delete data.createdAt;
  return data;
};
