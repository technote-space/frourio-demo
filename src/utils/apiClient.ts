import aspida from '@aspida/axios';
import api from '$/api/$api';

export const apiClient = api(aspida());
