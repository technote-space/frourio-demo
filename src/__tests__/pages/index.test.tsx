import React from 'react';
import dotenv from 'dotenv';
import Index from '~/pages/index';
import { render, waitFor, mockRouter } from '~/__tests__/utils';

dotenv.config({ path: 'server/.env' });

describe('Index page', () => {
  it('matches snapshot', async() => {
    const mock                      = mockRouter();
    const { container, asFragment } = render(<Index/>, {});

    await waitFor(() => {
      //
    }, { container: container as HTMLElement });

    expect(asFragment()).toMatchSnapshot();
    expect(mock.replace).toBeCalledTimes(1);
  });
});
