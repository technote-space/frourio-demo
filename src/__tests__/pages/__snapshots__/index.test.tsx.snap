// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrap-6"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-9 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-11 MuiTypography-h6"
        >
          予約システム
        </h6>
      </div>
    </header>
    <main
      class="makeStyles-main-7"
    >
      <div
        class="makeStyles-contents-8"
      >
        <div
          class="makeStyles-wrap-18"
        >
          <div
            class="makeStyles-wrap-19"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-22"
                  >
                    <label
                      class="MuiFormLabel-root Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value=""
                      />
                    </div>
                    <div
                      class="makeStyles-error-21"
                    />
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-22"
                  >
                    <label
                      class="MuiFormLabel-root Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-23"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-24 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-25"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Show
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="makeStyles-error-21"
                    />
                  </div>
                  <div
                    class="makeStyles-login-20"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-16"
    >
      <div
        class="makeStyles-wrap-17"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
